### ISSUE TITLE
Decoding Arrays from Query

### ISSUE THREAD

**Issue by thecheatah** [vapor/vapor#2138]
Trying to decode Arrays from a GET query string results in an unexpected fatalError. The behavior that does works is unintuitive and limits how you can name you GET parameters that can repeat multiple times.

### Steps to reproduce

Trying to decode an array of strings using the following
```swift
let emailsToSearch = try? request.query.get([String].self, at: "emailsToSearch")
```
results in a `fatalError` in [URLEncodedFormDecoder.swift line 124](https://github.com/vapor/vapor/blob/master/Sources/Vapor/URLEncodedForm/URLEncodedFormDecoder.swift#L124)
```swift
    func unkeyedContainer() throws -> UnkeyedDecodingContainer {
        guard let data = self.data else {
            return UnkeyedContainer(data: [], codingPath: self.codingPath)
        }
        switch data {
        case .array(let arr):
            return UnkeyedContainer(data: arr, codingPath: self.codingPath)
        default: fatalError()
        }
    }
```
when I send `/users?emailsToSearch=my@email.com` or `/users?emailsToSearch=my@email.com&emailsToSearch=another@email.com`

It does work when I send `/users?emailsToSearch[]=my@email.com` or `/users?emailsToSearch[]=my@email.com&emailsToSearch[]=another@email.com` however.

### Expected behavior

```swift
let emailsToSearch = try? request.query.get([String].self, at: "emailsToSearch")
```
should decode `/users?emailsToSearch=my@email.com` or `/users?emailsToSearch=my@email.com&emailsToSearch=another@email.com`

and 
```swift
let emailsToSearch = try? request.query.get([String].self, at: "emailsToSearch[]")
```
should decode 
`/users?emailsToSearch[]=my@email.com` or `/users?emailsToSearch[]=my@email.com&emailsToSearch[]=another@email.com`

### Actual behavior

Trying to decode an array using
```swift
let emailsToSearch = try? request.query.get([String].self, at: "emailsToSearch")
```
requires that GET arguments be named "emailsToSearch[]". This is unintuitive and assumes that array names should end with `[]`

### Environment

* Vapor Framework version: vapor 4.0.0-beta.3.1
* Vapor Toolbox version: 3.1.10 (Didn't use this to build the project using Xcode)
* OS version: Mac OS 10.15.2

**Comment by ckd** [comment]
As far as I know, this is not covered by any RFC. The current behavior is consistent with how some other languages/frameworks operate (PHP comes to mind).

**Comment by thecheatah** [comment]
If it’s intentional then that’s fine, but it should be documented and it shouldn’t crash. Ideally the “crash” should indicate the correct way to use arrays with query parameters.

I still don’t like the idea of not being able to process repeating GET parameters unless they end with [].

**Comment by Joannis** [comment]
This has been solved in #2177
